### PR TITLE
Add install step + example + test

### PR DESCRIPTION
**Pull request by Warchant**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,3 +60,11 @@ jobs:
       env:
         CTEST_OUTPUT_ON_FAILURE: 1
       run: cmake --build build --target test
+#    - name: install & test
```
Comment (Harrm) [inline .github/workflows/ci.yml]: Why are these lines commented out?

File: cmake/install.cmake [hunk 2]
```diff
@@ -1,46 +1,74 @@
-function(install_deps_headers)
-  install(DIRECTORY ${MICROSOFT.GSL_ROOT}/include/gsl
-      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+#function(install_deps_headers)
```
Comment (Harrm) [inline cmake/install.cmake]: And these?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Warchant) [reply .github/workflows/ci.yml]: added comment

Comment (Warchant) [reply .github/workflows/ci.yml]: added comment above 🙃

File: example/00-install/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+HunterGate(
+    URL "https://github.com/cpp-pm/hunter/archive/v0.23.216.tar.gz"
+    SHA1 "a492a1879bee60c29f7af17ef56b45f0d7766241"
+)
+
+
+project(00-install C CXX)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (masterjedy) [inline example/00-install/CMakeLists.txt]: doesn't it prevent libp2p from being installed from archive using hunter?

**Pull request merged**